bika/lims/browser/analysisrequest/analysisrequests.py

```diff
@@ -132,12 +138,12 @@ def __init__(self, context, request):
             'getSampler': {'title': _('Sampler'),
                            'toggle': SamplingWorkflowEnabled},
             'getDatePreserved': {'title': _('Date Preserved'),
-                                 'toggle': user_is_preserver,
+                                 'toggle': self.user_is_preserver,
```

Comment:
The value for `self.user_is_preserver` is assigned in `__call__`, so the value here will always be False. The toggle thing  assignment should be done in `__call__` . I see in the previous version, `self.user_is_preserver` was assigned in `__init__` though (afak, role checking doesn't work properly in `__init__`, so don't know how it could work before).